src/main/java/lotto/controller/LottoController.java

```diff
@@ -1,46 +1,31 @@
 package lotto.controller;
 
 import java.util.List;
-import lotto.domain.Lotto;
+import lotto.domain.LottoBuyMoney;
 import lotto.domain.Lottos;
-import lotto.domain.vo.Money;
-import lotto.domain.vo.Number;
-import lotto.domain.Rank;
-import lotto.domain.Store;
-import lotto.domain.WinnerLotto;
-import lotto.view.InputView;
-import lotto.view.OutputView;
+import lotto.domain.Statistics;
+import lotto.dto.LottosResult;
+import lotto.dto.StatisticsResult;
+import lotto.service.LottoService;
 
 public class LottoController {
 
-    public void run() {
-        Money inputMoney = InputView.inputMoney();
+    private final LottoService lottoService;
 
-        Lottos lottos = new Lottos(buyLottos(inputMoney));
-        OutputView.printLottos(lottos.getLottos());
-
-        List<Rank> ranks = lottos.match(createWinnerLotto(winnerNumbers(), bonusNumber()));
-        OutputView.printRanks(ranks);
-
-        Money totalReward = Rank.calculateReward(ranks);
-        OutputView.printRate(totalReward, inputMoney);
-    }
-
-    private List<Lotto> buyLottos(Money money) {
-        Store store = new Store(money);
-        return store.buyLottos();
+    public LottoController(LottoService lottoService) {
+        this.lottoService = lottoService;
     }
 
-    private WinnerLotto createWinnerLotto(List<Number> winnerNumbers, Number bonusNumber) {
-        return new WinnerLotto(new Lotto(winnerNumbers), bonusNumber);
+    public LottosResult buyLotto(int inputMoney, int manualAmount, List<List<Integer>> manualNumbers) {
+        LottoBuyMoney lottoBuyMoney = new LottoBuyMoney(inputMoney);
+        int autoLottoAmount = lottoBuyMoney.countAutoAmountByManualAmount(manualAmount);
+        Lottos lottos = lottoService.createLottos(autoLottoAmount, manualNumbers);
+        return new LottosResult(lottos.getLottos(), autoLottoAmount, manualAmount);
     }
 
-    private List<Number> winnerNumbers() {
-        return InputView.inputWinnerNumbers();
+    public StatisticsResult match(List<List<Integer>> lottos, List<Integer> winnerNumbers, int bonusNumber, int inputMoney) {
```

Comment:
request dto를 만들어서 매개변수 수를 줄이고 응집도를 높일 수도 있습니다!
```suggestion
    public StatisticsResult match(StatisticsRequest request) {
```